src/compiler.ts

```diff
@@ -1106,6 +1110,20 @@ export class Compiler extends DiagnosticEmitter {
 
   // === Globals ==================================================================================
 
+  /** Tries to compile a global variable lazily. */
+  compileGlobalLazy(global: Global, reportNode: Node): bool {
+    if (global.is(CommonFlags.Compiled)) return !global.is(CommonFlags.Errored);
+    if (global.hasAnyDecorator(DecoratorFlags.Lazy | DecoratorFlags.Builtin) || global.is(CommonFlags.Ambient)) {
```

Comment:
If `builtin` decorator can allow to compile global lazy, this case also needs to be considered when switch flow

Reply:
otherwise the result of this code is still incorrect

```ts
// node_modules/t/index.ts
class T {
    a: i32 = 1;
}
@builtin @global
let t = new T();

// index.ts
class T {}
let b = t;

import {} from "t/index";
```

Reply:
Should not be a problem, since the compiler is in control of `@builtin`s. The example diagnoses `AS212: Decorator '@builtin' is not valid here.`. Currently, this is really only used for variables like `NaN` (context-sensitive constant), `__heap_base` (compiler-injected constant) and such which do not have side-effects.

Reply:
but everyone can use this decorator and mislead compiler. Is it possible to find a solution to avoid it? For example, forbidden builtin decorator for all non-std code even in library.

Reply:
I would agree that we need a better story for the `@builtin` decorator, and how issues with it are diagnosed, as it's currently inconsistent. As you say, it should not be allowed that one can use it for something it is not intended for. From looking at the code again, it's merely a shortcut currently so the full builtin machinery isn't required for variables. Need to check if there's anything simple that can be done about it here.

Reply:
Addressed in #2636, PTAL